app/js/streaming/ProtectionExtensions.js

```diff
@@ -84,41 +92,86 @@ MediaPlayer.dependencies.ProtectionExtensions.prototype = {
         if (null !== cdmData) {
             return mediaKeys.createSession(mediaCodec, initData, cdmData);
         }
+
         return mediaKeys.createSession(mediaCodec, initData);
     },
 
     getKeySystems: function (protectionData) {
         var self = this,
-            _protectionData = protectionData,
-            getLAUrl = function (laUrl, keysystem) {
-                if (protectionData[keysystem] !== undefined) {
-                    if (protectionData[keysystem].laUrl !== null && protectionData[keysystem].laUrl !== '') {
-                        return protectionData[keysystem].laUrl;
+            _stringUtils = self.stringMethods,
+            _protectionData = {
+                licenseRequest: {
+                    'com.microsoft.playready': {
+                        cdmData: null,
+                        laUrl: null,
+                        headers: null
+                    },
+                    'com.widevine.alpha': {
+                        cdmData: null,
+                        laUrl: null,
+                        headers: null
                     }
                 }
-                return laUrl;
             },
-            playreadyGetUpdate = function (event) {
-                var decodedChallenge = null,
+            getUpdate = function (event, keySystem, laUrl, headers, message) {
+                var xhr = new XMLHttpRequest();
+
+                xhr.open('POST', laUrl, true);
```

Comment:
Hey @greg80303,

The scenario that Mikhail fixed there is handled differently in my branch (lines 102-115 & 292-294) but already handled I believe.